pkg/trustedresources/verify.go

```diff
@@ -78,19 +80,20 @@ func VerifyTask(ctx context.Context, taskObj *v1beta1.Task, k8s kubernetes.Inter
 		if errors.Is(err, ErrNoMatchedPolicies) {
 			switch config.GetVerificationNoMatchPolicy(ctx) {
 			case config.IgnoreNoMatchPolicy:
-				return nil
+				return VerificationResult{VerificationResultType: VerificationSkip}
 			case config.WarnNoMatchPolicy:
 				logger := logging.FromContext(ctx)
-				logger.Warnf("failed to get matched policies: %v", err)
-				return nil
+				warning := fmt.Errorf("failed to get matched policies: %w", err)
+				logger.Warnf(warning.Error())
+				return VerificationResult{VerificationResultType: VerificationWarn, Err: warning}
 			}
 		}
-		return fmt.Errorf("failed to get matched policies: %w", err)
+		return VerificationResult{VerificationResultType: VerificationError, Err: fmt.Errorf("failed to get matched policies: %w", err)}
 	}
```

Comment:
I feel that having this logic after getting matched policies inside VerifyTask has 2 problems: 1). it makes the steps in function VerifyTask look like too long and hard to understand. 2). duplicated code exists in VerifyPipeline.

2 suggestions:
- Can we nest this logic checking config into `getMatchedPolicies` function itself by passing the context to it? We can refactor `getMatchedPolicies` so that it returns 2 things: `[]*v1alpha1.VerificationPolicy` and `*VerificationResult`? If returned `*VerificationResult` is not nil, we just return the result from inside VerifyTask. If nil, proceed to next step.
- Also, can we just rename `getMatchedPolicies ` to `matchPolicies`? see https://google.github.io/styleguide/go/decisions#getters. Perhaps also add a doc string to this function though it's not exported b/c it has a bit more logics.

This will help mitigate the 2 problems mentioned above.

Reply:
Thanks! This is a reasonable suggestion.  I'm planing to merge these 2 functions into 1. Maybe we could leave this non-related change in the refactoring PR? 

If think we have docstrings for `getMatchedPolicies`.

Reply:
For the naming, I think the doc you linked is for `getter`, which is used as a member function to return some data from the struct, I think it is different from what we have here?

Reply:
Doing this in another PR SGTM.

The naming standard is for both functions and member methods. "Function and method names should not use a Get or get prefix".